src/downloadManager/__init__.py

```diff
@@ -69,7 +70,16 @@ def _finish(self):
 
     def run(self):
         self._running = True
+        token = Settings.get("oauth/token")['access_token']
+
+        o = urlparse(self.addr)
+        query = parse_qs(o.query)
+
         req = QNetworkRequest(QUrl(self.addr))
+        if 'verify' in query:
+            req = QNetworkRequest(QUrl(o._replace(query=None).geturl()))
+            req.setRawHeader(b'Verify', query['verify'][0].encode())
```

Comment:
Unfortunately, this doesn't work for me, because `parse_qs` decodes percent-encoded sequences and the server can't validate this correctly (unless we manually mess with the encoding).
What about using PyQt5's `QUrlQuery` for this? (It [doesn't decode everything](https://doc.qt.io/qt-5/qurlquery.html#handling-of-spaces-and-plus))
something like this:
```Python
qurl = QUrl(self.addr)
query = QUrlQuery(qurl)
qurl.setQuery(QUrlQuery())
req = QNetworkRequest(qurl)
if not query.isEmpty():
    for query_key, query_value in query.queryItems():
        req.setRawHeader(query_key.encode(), query_value.encode())
        req.setRawHeader(b'Authorization', b'Bearer ' + token.encode())
```